apollo-router/src/services/layers/persisted_queries/mod.rs

```diff
@@ -15,8 +15,8 @@ use crate::graphql::Error as GraphQLError;
 use crate::services::SupergraphRequest;
 use crate::services::SupergraphResponse;
 use crate::Configuration;
-use crate::UplinkConfig;
 
+const SANDBOX_INTROSPECTION_QUERY: &str = "\n    query IntrospectionQuery {\n      __schema {\n        \n        queryType { name }\n        mutationType { name }\n        subscriptionType { name }\n        types {\n          ...FullType\n        }\n        directives {\n          name\n          description\n          \n          locations\n          args {\n            ...InputValue\n          }\n        }\n      }\n    }\n\n    fragment FullType on __Type {\n      kind\n      name\n      description\n      \n      fields(includeDeprecated: true) {\n        name\n        description\n        args {\n          ...InputValue\n        }\n        type {\n          ...TypeRef\n        }\n        isDeprecated\n        deprecationReason\n      }\n      inputFields {\n        ...InputValue\n      }\n      interfaces {\n        ...TypeRef\n      }\n      enumValues(includeDeprecated: true) {\n        name\n        description\n        isDeprecated\n        deprecationReason\n      }\n      possibleTypes {\n        ...TypeRef\n      }\n    }\n\n    fragment InputValue on __InputValue {\n      name\n      description\n      type { ...TypeRef }\n      defaultValue\n      \n      \n    }\n\n    fragment TypeRef on __Type {\n      kind\n      name\n      ofType {\n        kind\n        name\n        ofType {\n          kind\n          name\n          ofType {\n            kind\n            name\n            ofType {\n              kind\n              name\n              ofType {\n                kind\n                name\n                ofType {\n                  kind\n                  name\n                  ofType {\n                    kind\n                    name\n                  }\n                }\n              }\n            }\n          }\n        }\n      }\n    }\n  ";
```

Comment:
As the apollo compiler is available why not use this?

Reply:
Long term I think we want to use the compiler, but we wanted to get this in for preview, so this is here as a stopgap (hence the `// TODO` comments. 

We have two tickets here: one for the [short term](https://github.com/mdg-private/planning/issues/2445) (which this PR solves) and one for the [long term](https://github.com/mdg-private/planning/issues/2446). The long term fix will require using the compiler as you note, but it will first require fixing [this bug in the existing introspection logic](https://github.com/apollographql/router/issues/2789) that, if used in its current state, would allow malicious unsafelisted queries to masquerade as an introspection query.